supporting_content/attributes/discount_handling.md

```diff
@@ -0,0 +1,117 @@
+# Discount Types
+
```

Comment:
```suggestion

## Example implementation of the Discount Handling attribute

Let's say you have 3 discounts:

1. **10% negotiated** discount
2. **10% bundled** discount
3. **$1/hour spend commitment** purchased for **$0.50/hour**

For the sake of this example, let's say the negotiated and bundled discounts cap out and don't apply to all resources to demonstrate what happens with a partially undiscounted amount.

All examples are $0.05/minute for hourly data...

| Scenario | Resource | Qty | List | OnDemand | Billed | Effective |
|-----------|-----------|------|-----|--------------|-------|-----------|
| No discounts | Foo0 | 60 | $3.00 | $3.00 | $3.00 | $3.00 |
| Negotiated only | Foo1 |  60 | $3.00 | $2.70 | $2.70 | $2.70 |
| Commitment only | Foo2 |  20 | $1.00 | $1.00 | $0.00 | $0.50 |
| Uncommitted part | Foo2 | 40 | $2.00 | $2.00 | $2.00 | $2.00 |
| Commitment only | Foo3 | 10 | $0.50 | $0.50 | $0.00 | $0.25 |
| Unused commitment | | | $0.00 | $0.00 | $0.00 | $0.25 |
| Negotiated +&nbsp;committed | Foo4 | 22.22 | $1.11 | $1.00 | $0.00 | $0.50 |
| Negotiated +&nbsp;uncommitted | Foo4 | 37.78 | $1.89 | $1.70 | $1.70 | $1.70 |
| All discounts<sup>1</sup> | Foo5 | 24.69 | $1.11 | $1.00 | $0.00 | $0.50 |
| Negotiated +&nbsp;bundled | Foo5 | 5.31 | $0.24 | $2.15 | $2.15 | $2.15 |
| Negotiated only | Foo5 | 15 | $0.75 | $0.67 | $0.67 | $0.67 |
| All discounts capped out | Foo5 | 15 | $0.75 | $0.75 | $0.75 | $0.75 |

<sup>_1. As we've defined SKUs, bundled discounts would be applied to the list price, so that's why it doesn't look like there's a discount applied compared to Foo4._</sup>

Explanation of the unused commitment row:

1. The **Foo3** resource only used **10min @ $0.05/min**.
2. This is **$0.50** list _and_ on-demand since the negotiated discount doesn't apply on this row for the sake of the example (see Foo5 for stacked discounts).
3. The billed cost is nothing, since this $0.50 was covered by the spend commitment.
4. Since the spend commitment was for $1/hour, this means there's $0.50 left that was not used.
5. The "Unused commitment" row is for that unused amount.
   - List, billed, and on-demand costs are nothing since there was no usage and nothing was billed.
   - Effective cost is the unused amount of the spend commitment ($0.50 in this case).
   - Since spend commitments don't have a unit or a price, they won't have a quantity, so the unused amount doesn't have a quantity.
   - The only thing it has is the unused amount, which is the effective price per hour ($0.50), minus the used portion ($0.25).
   - **_Why $0.25?_**
     - The commitment was $1/hour with a 50% discount per hour.
     - This means the effective/amortized cost per hour is $0.50.
     - The on-demand cost was $0.50. After applying the amortized price, it's 50% off or $0.25.
     - Since the on-demand cost was only $0.50, there's still another $0.50 left unused.
     - And when you apply the 50% discount on the unused portion, you get $0.25.
     
```